Services/BrokerService/Controllers/BrokerController.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using BrokerServices;
+using BrokerServices.Utils;
+using Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace BrokerService.Controllers
+{
+    [Route("[controller]")]
+    [Controller]
+    public class BrokerController : ControllerBase
+    {
+        private readonly ILogger<BrokerController> logger;
+
+        public BrokerController([FromServices] ILogger<BrokerController> logger)
+        {
+            this.logger = logger;
+        }
+
+        [Route("getAllCurrencies")]
+        [HttpPost]
+        public List<IMarketInstrument> GetAllCurrencies([FromBody] BankType bankType)
+        {
+            return BrokerFactory.Create(bankType,logger).GetAllCurrencies();
+        }
+
+        [Route("getCurrency")]
+        [HttpPost]
```

Comment:
Why posts?

Reply:
Because we need a body

Reply:
Why not use query params and change this to HttpGet?

Reply:
Also in other http queries

Reply:
Also in other http requests.

Reply:
Also in other http queries